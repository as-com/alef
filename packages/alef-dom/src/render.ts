import { renderToElement } from "alef-tools";

import createDOMInterface from "./DOMInterface";
import initStyleNodes from "./initStyleNodes";

import DOMRenderer from "../../../types/DOMRenderer";
import DOMNode from "../../../types/DOMNode";

let warnDeprecated = false;

export default function render(
	renderer: DOMRenderer,
	mountNode: DOMNode
): void {
	// legacy support for single stylesheets
	// will be removed soon
	if (mountNode) {
		if (!warnDeprecated) {
			// TODO: warn
			warnDeprecated = true;
		}

		renderToElement(renderer, mountNode);
	} else {
		initStyleNodes(renderer);
		const updateInterface = createDOMInterface(renderer);
		renderer.subscribe(updateInterface);
	}
}
