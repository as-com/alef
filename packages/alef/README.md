# fela

NPM package to get access to Fela's API.<br>
Check out the [Fela repository](https://github.com/rofrischmann/fela) for further information.
