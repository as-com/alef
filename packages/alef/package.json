{
  "name": "alef",
  "version": "5.0.1",
  "description": "Universal, Dynamic & High-Performance Styling in JavaScript",
  "typings": "index.d.ts",
  "main": "lib/index.js",
  "module": "es/index.js",
  "jsnext:main": "es/index.js",
  "files": [
    "LICENSE",
    "README.md",
    "index.d.ts",
    "lib/**",
    "es/**"
  ],
  "repository": "https://github.com/as-com/alef/",
  "keywords": [
    "alef",
    "dynamic styling",
    "stylesheet",
    "css",
    "styling",
    "cssinjs"
  ],
  "author": "Robin Frischmann",
  "license": "MIT",
  "devDependencies": {
    "alef-monolithic": "^5.0.1"
  },
  "dependencies": {
    "alef-tools": "^5.0.1",
    "alef-utils": "^5.0.1",
    "css-in-js-utils": "^1.0.3"
  }
}
