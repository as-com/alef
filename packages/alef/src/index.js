/* @flow */
import createRenderer from './createRenderer'
import combineRules from './combineRules'
import enhance from './enhance'
import connectFactory from './bindings/connectFactory'
import createComponentFactory from './bindings/createComponentFactory'

export {
  createRenderer,
  combineRules,
  enhance,
  connectFactory,
  createComponentFactory
}
