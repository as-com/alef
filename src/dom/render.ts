import { renderToElement } from "alef-tools";

import createDOMInterface from "./DOMInterface";
import initStyleNodes from "./initStyleNodes";

import { DOMRenderer } from "../../types/DOMRenderer";
import { DOMNode } from "../../types/DOMNode";

let warnDeprecated = false;

export default function render(
	renderer: DOMRenderer
): void {
	initStyleNodes(renderer);
	const updateInterface = createDOMInterface(renderer);
	renderer.subscribe(updateInterface);
}
