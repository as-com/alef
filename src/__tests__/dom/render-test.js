import { html as beautify } from "js-beautify";
import { createRenderer } from "alef";
import render from "../../dom/render";

describe("render", () => {
	it("should create style nodes and render CSS rules", () => {
		const renderer = createRenderer();
		renderer.renderRule(() => ({
			backgroundColor: "red",
			color: "blue"
		}));
		renderer.renderKeyframe(() => ({
			"0%": { color: "yellow" },
			"100%": { color: "orange" }
		}));
		renderer.renderFont("Lato", ["../Lato.ttf"], { fontWeight: 300 });
		render(renderer);
		expect(
			beautify(document.documentElement.outerHTML, { indent_size: 2 })
		).toMatchSnapshot();
	});
});
