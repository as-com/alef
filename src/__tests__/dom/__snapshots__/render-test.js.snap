// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should create style nodes and render CSS rules 1`] = `
"<html>

<head>
  <style data-alef-type=\\"FONT\\" type=\\"text/css\\">
    @font-face {
      font-weight: 300;
      src: url('../Lato.ttf') format('truetype');
      font-family: \\"Lato\\"
    }
  </style>
  <style data-alef-type=\\"KEYFRAME\\" type=\\"text/css\\">
    @-webkit-keyframes k1 {
      0% {
        color: yellow
      }
      100% {
        color: orange
      }
    }

    @-moz-keyframes k1 {
      0% {
        color: yellow
      }
      100% {
        color: orange
      }
    }

    @keyframes k1 {
      0% {
        color: yellow
      }
      100% {
        color: orange
      }
    }
  </style>
  <style data-alef-type=\\"RULE\\" type=\\"text/css\\">
    .a {
      background-color: red
    }

    .b {
      color: blue
    }
  </style>
</head>

<body></body>

</html>"
`;
